### ISSUE TITLE
Child list will not map to a parent with a nullable ID

### ISSUE THREAD

**Issue by thespatt** [jonwagner/Insight.Database#265]
When including an ID Func and an into Func, mapping into a parent with a nullable ID fails.
This test fails:
```c#
class Parent
{
    public int? ID;
    public List<int> Children;
}
[Test]
public void ChildMapsToParentWithNullableId()
{
    var returns = Query.Returns(Some<Parent>.Records)
        .ThenChildren(Some<int>.Records, x => x.ID, (p, r) => p.Children = r.Select(x => x).ToList());
    var results = Connection().QuerySql("SELECT ID=1; SELECT ParentID=1, ID=2;", null, returns);
    Assert.AreEqual(1, results[0].ID);
    Assert.AreEqual(2, results[0].Children[0]);
}
```

**Comment by jonwagner** [comment]
Ooh a fun IL function mapping bug. Whoever gets this one will have a good time.

**Comment by Jaxelr** [comment]
There is an issue with the conversion of the Nullable int at this specific point of the ChildMapper in here https://github.com/jonwagner/Insight.Database/blob/master/Insight.Database/Structure/ChildMapper.cs#L79 

As expected nullables require certain specific behavior using the Nullable.GetUnderlyingType operation to avoid this issue.

**Comment by jonwagner** [comment]
Fixed in v5.2.8
